### PR TITLE
EWPP-2701: Add new optional fields to corporate forms.

### PR DESCRIPTION
**Pull request by yenyasinn**

### CODE REVIEW

File: tests/src/FunctionalJavascript/CorporateContactFormTest.php [hunk 1]
```diff
@@ -172,6 +179,22 @@ public function testCorporateContactForm(): void {
     // Assert fields are now visible.
     $this->assertFieldsVisible(TRUE);
 
+    // Assert alternative contact language is active only when preferred
+    // contact language field is active.
```
Comment (nagyad) [inline tests/src/FunctionalJavascript/CorporateContactFormTest.php:183]: We should test that it is not available for non-corporate forms.

Comment (nagyad) [reply tests/src/FunctionalJavascript/CorporateContactFormTest.php:183]: Also I don't see the case when we have Preferred checkbox saved the Alternative checkbox is not disabled on page load.

File: oe_contact_forms.module [hunk 2]
```diff
@@ -468,6 +473,23 @@ function _oe_contact_forms_email_validate(array &$form, FormStateInterface $form
...
+ * @param Drupal\Core\Form\FormStateInterface $form_state
+ *   The form state.
+ */
+function _oe_contact_forms_alternative_contact_language_validate(array &$form, FormStateInterface $form_state): void {
+  $values = &$form_state->getValue('corporate_fields');
+  if (empty($values['optional_fields']['oe_preferred_language'])) {
+    // Uncheck alternative contact language if preferred language is unchecked.
+    $values['optional_fields']['oe_alternative_language'] = '';
+    $form_state->setValue('corporate_fields', $values);
+  }
```
Comment (nagyad) [inline oe_contact_forms.module:490]: Why we do this in a validate, we are not validating anything here we are setting the value of a field based on another field.

File: oe_contact_forms.post_update.php [hunk 3]
```diff
@@ -14,3 +14,67 @@ function oe_contact_forms_post_update_00001(): void {
...
+    ->setDisplayOptions('view', [
+      'weight' => 0,
+      'settings' => [
+        'link' => FALSE,
+      ],
+    ]);
+  $definition_update_manager->installFieldStorageDefinition('oe_preferred_language', 'contact_message', 'oe_contact_forms', $storage_definition);
+
+  $storage_definition = BaseFieldDefinition::create('skos_concept_entity_reference')
+    ->setLabel(t('Alternative contact language'))
```
Comment (nagyad) [inline oe_contact_forms.post_update.php:56]: In the base_field_info you avoided the code duplication, here why not? They are the same field with different labels.

File: src/Plugin/ConceptSubset/ContactLanguages.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+ *   }
+ * )
+ */
+class ContactLanguages extends ConceptSubsetPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function alterQuery(QueryInterface $query, $match_operator, array $concept_schemes = [], string $match = NULL): void {
+    $languages = [
```
Comment (nagyad) [inline src/Plugin/ConceptSubset/ContactLanguages.php:28]: This should be tested against a fixture containing the languages it has to contain.

File: js/related_checkboxes.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+/**
+ * @file
+ * "Related checkboxes" library file.
+ */
+(function (Drupal, $) {
+  /**
+   * Sets state of "Alternative contact language" checkbox based on "Preferred contact language" state.
```
Comment (nagyad) [inline js/related_checkboxes.js:7]: Let's provide a proper method annotation here. Look for examples in core or ewcms.

**Review by nagyad — CHANGES_REQUESTED** [review]

**Review by nagyad — DISMISSED** [review]

**Review by sergepavle — APPROVED** [review]

**Review by nagyad — APPROVED** [review]

**Pull request merged**